City-of-Helsinki/employment-services-ui
# Thf 502 nav changes

Renewed the navigation.

**New functions/ what is been changed:**

- Created 3 new components navigation, mobileNavigation and languageSelect. 
- New react hook for blurring the drop-down menu. 
- Changed breadcrumbs to handle also so, uk and ru languages.
- Footer fallback menu for so, uk and ru languages. Footer is shown in English with these languages.
- Nav menu rows are hidden if language is something else than primary languages.
- Search button is only shown in primary language pages (fi, sv, en)
- If a user tries to change language from language select and the page doesn't have translation the user is moved to the front page of the language the user tried to choose. 
- New menu handling for other languages
- Current secondary language is shown on the language selection. (If we are example in uk front page the language selection shows Ukraine with primary languages) 
- The users are redirected from /uk, /so and /so to the actual landing pages.

**Notes for code review:**
- I had to create completely 2 different component for navigation. HDS navigation don't let put any new components in the navigation action section on small screen. All components except search and user are hidden from navigation when put between navigation actions. If I put components outside the action component the added components ended to be last place on the navigation. It didn’t even let create new custom components with just search and user components and use it inside action the navigation action component.
- In the big screen navigation (navigation.tsx) I created completely new order for the navigation items and language select is with the navigation actions. 
- Mobile screen navigation follows the same old layout. Only language select is moved to new place under the main navigation. Other languages is next to the main language select.
- I had to create front-page link for the breadcrumbs with code because Next.js redirection didn't redirect from actual link '/uk' etc. on the page. With primary languages it works with '/' because the actual front page is behind the link but for secondary languages had to write the actual link to the front page. Breadcrumb.tsx rows 45-62

**Important:**

Need to do before deploy --> change the redirection links and helper front page link to the real ones. 

**Questions 🥇 ** 
In the file LanguageSelect React best practices would be create 2 different component out of the main language select and the other drop-down menu and combine the components into one in this file. What you think, should I do small components or leave it be? I would prefer 2, but the project structure isn’t maybe best for it.  


**What to test:**

You need to have Drupal back end with ability to handler ru, so and uk. Fall back footer created with English links.

- Test that the layout hasn’t change because of the changes. 
- Main nav drop-downs works as before
- The new global drop-down works -> You can toggle the drop-down menu -> The menu open to the right place - The links are working -> and on blur the drop-down menu closes. 
- Also test accessibility - it should work almost same as hel.fi -> when you enter to the main languages it says the first language and ask you to "Select site language" when you are on the active language it tell you current Item. -> when you enter the other languages button the reeder tells you are on "Open Information in other languages menu"
- Create landing page for ru, so and so for testing purpose create page with titles below
- uk: 'ukrainian-landing-page',
  so: 'somali-landing-page',
  ru: 'russian-landing-page',
- The new landing pages should not have search or nav rows on the page.
- Create at least one of the new landing pages basic page
- Try to toggle front pages with the language select
- uk, ru and so should redirected to the landing pages you just created
- Go one of the basic pages you created with new languages 
- test the breadcrumbs -> you should end to the front page of the language the page belongs to.
- Go back to the basic page you were before and try to change language (choose language that doesn't have translation) you should end up to the front page of the language you tried to switch.
